ingest/expression_files/expression_files.py

```diff
@@ -100,78 +129,45 @@ def check_unique_cells(cell_names: List, study_id, client):
         ]
         dupes = set(existing_cells) & set(cell_names)
         if len(dupes) > 0:
-            error_string = f"Expression file contains {len(dupes)} cells that also exist in another expression file. "
+            error_string = (
+                f"Expression file contains {len(dupes)} cells "
+                "that also exist in another expression file."
+            )
+
             # add the first 3 duplicates to the error message
             error_string += f'Duplicates include {", ".join(list(dupes)[:3])}'
             raise ValueError(error_string)
         return True
 
-    def set_data_array_cells(self, values: List, linear_data_id):
-        """Sets DataArray for cells that were observed in an
-        expression matrix."""
-        for model in DataArray(
-            f"{self.cluster_name} Cells",
-            self.cluster_name,
-            "cells",
-            values,
-            "Study",
-            linear_data_id,
-            self.study_id,
-            self.study_file_id,
-        ).get_data_array():
-            yield model
-
-    def set_data_array_gene_cell_names(
-        self, name: str, linear_data_id: str, values: List
-    ):
-        """Sets DataArray for cell names associated to a single gene. This
-        DataArray contains cell names that had significant (i.e. non-zero)
-        expression for a gene.
-        """
-        for model in DataArray(
-            f"{name} Cells",
-            self.cluster_name,
-            "cells",
-            values,
-            "Gene",
-            linear_data_id,
-            self.study_id,
-            self.study_file_id,
-        ).get_data_array():
-            yield model
-
-    def set_data_array_gene_expression_values(
-        self, name: str, linear_data_id: str, values: List
-    ):
+    @staticmethod
+    def create_data_array(
+        *ignore,
+        # keyword arguments
+        name: str,
+        cluster_name: str,
+        array_type: str,
+        values: List,
+        linear_data_type: str,
+        linear_data_id,
+        study_id,
+        study_file_id,
+    ) -> Generator:
         """
-        Sets DataArray for expression values for a gene. This is an array of
-        significant (i.e. non-zero) expression values for a gene.
+        Sets data array.
+        Only allows keyword arguments
         """
-        for model in DataArray(
-            f"{name} Expression",
-            self.cluster_name,
-            "expression",
-            values,
-            "Gene",
-            linear_data_id,
-            self.study_id,
-            self.study_file_id,
-        ).get_data_array():
+        fn_kwargs = copy.copy(locals())
+        # Positional arguments passed in
+        if ignore:
+            raise TypeError("Positional arguments are not accepted.")
+        del fn_kwargs["ignore"]
+        for model in DataArray(**fn_kwargs).get_data_array():
```

Comment:
I don't want to touch get_data_array() right now because all file types/metadata convention use this method. Testing for other file types/metadata convention aren't as robust as for expression files. I have no doubt that I'll miss something and cause an error in prod.

Reply:
yes, that's we aren't merging/releasing this change for the dense parsing that's going out today.   I only see 3 total mentions of this function in the current codebase--that's not a risky change--and it's also something that any smoke test would catch b/c the ingest  for that type would fail altogether if you missed a type.  Use your editor's "find/replace all" to look for all instances and change them.